active/0000-engines.md

```diff
@@ -0,0 +1,411 @@
+        ]
+      }
+    }
+  }
+});
+
+export default App;
+```
+
+When engines are instantiated, the listed dependencies will be looked up on
+the parent and made accessible within the engine.
+
+Note that the `engines` declaration provides further space to define
+characteristics about an engine, such as whether it should be eager or
+lazy-loaded, URLs for manifest files, etc.
+
+# Drawbacks
+
+This RFC introduces the new concept of engines, which increases the
+learning curve of the framework. However, I believe this issue is
+mitigated by the fact that engines are an opt-in packaging around
+existing concepts.
+
+In the end, I believe that "engines" are just a small API for composing
+existing concepts. And they can be introduced at the top of the
+conceptual ladder, once users are comfortable with the basics of Ember,
+and only for those working on large teams or distributing addons.
+
+# Alternatives
+
+Several incomplete alternatives are discussed in the Motivations section above.
+
+I know of no alternatives being discussed in the Ember community that meet the
+same needs as engines; namely, for development _and_ run-time isolation.
+
+# Unresolved questions
+
+## Non-CLI Users
+
+This RFC assumes Ember CLI. I would prefer to prove this out in Ember
+CLI before locking down the public APIs/hooks the router exposes for
+locating and mounting engines. Once this is done, however, we should
+expose and document those hooks so users who cannot use Ember CLI for
+whatever reason can still take advantage of composability.
+
+## Declarative dependencies
+
+The initial scope of declarative dependency sharing is limited in scope to
+services. Should other types of dependencies be declaratively shareable?
+Should addons be the recommended path to share all other dependencies?
+
+## Async mounting of route-less engines
+
+`Route#renderTemplate` is called synchronously, although `Route#mount` should
+surely be async. How async mounting is represented in the route lifecycle is
+TBD. A solution isn't proposed here because the problem is shared by routable
+and async components, and a common solution should be reached.
+
+## Lazy loading manifests
+
+In order to facilitate lazy loading of engines, we will need to determine a
+structure for manifest files that contain an engine's assets. Furthermore, an
+application will need to be configurable with URLs for these manifests.
+
+It's likely that an engine's routing map will always be needed at the time of
+application deployment. Allowing lazy loading of routing maps would prevent the
+formation of any links from a parent application into an engine's routes.
+
+When developed in isolation as addons, engines will have their own sets of
+dependencies. These dependencies will be treated like any other addons when
+engines are deployed together with an application. However, in order to support
+lazy loading, it would be ideal to dedupe dependencies in order to create a lean
+and conflict-free asset manifest.
```

Comment:
Non-naive solutions to shared dependency problems are both difficult and important. These problems seem larger than the engines RFC and even ember itself. 

Like you, I'm glad that we have some veterans of similar efforts in our community, and would of course be grateful if @wycats and/or @indirect took the lead here. No pressure though ;)